### PR TITLE
Add native Expo Google API integration for social login

### PR DESCRIPTION
**Pull request by BaloPhilip**

### CODE REVIEW

File: modules/user/server-ts/social/googleExpo/resolvers.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    password: id,
+    isActive: true
+  });
+};
+
+const createGoogleOAuth = async user => User.createGoogleOAuth(user);
+
+export default () => ({
+  Mutation: {
+    async googleExpoLogin(
```
Comment (evgenijolshanskij) [inline modules/user/server-ts/social/googleExpo/resolvers.js]: Very bad approach. Every client is able to send a request to the server passing any `id | name | email` and successfully being registered in the system. Without any verifications. User **must** be authenticated on the server side.

File: modules/authentication/client-react/helpers/index.js [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+  } = settings;
+
+  if ((isLogin && isExpoLogin) || isLogin) {
+    return () => login();
+  } else {
+    return client => expoLogin(client);
+  }
+};
+
+export const saveTokens = async ({ accessToken, refreshToken }) => {
```
Comment (evgenijolshanskij) [inline modules/authentication/client-react/helpers/index.js]: We already have a declared `saveTokens` function in the App.

File: config/auth.js [hunk 3]
```diff
@@ -52,6 +52,11 @@ export default {
       clientSecret: process.env.GOOGLE_CLIENTSECRET,
       callbackURL: '/auth/google/callback',
       scope: ['https://www.googleapis.com/auth/userinfo.email', 'https://www.googleapis.com/auth/userinfo.profile']
+    },
+    googleExpo: {
```
Comment (evgenijolshanskij) [inline config/auth.js]: All the settings related to `google` should be under the `google` scope.

File: modules/user/client-react/components/LoginView.web.jsx [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
 import LoginForm from './LoginForm';
 
-const LoginView = ({ onSubmit, t, isRegistered, hideModal }) => {
+const LoginView = ({ onSubmit, t, isRegistered, hideModal, history, client }) => {
+  const {
+    location: { search }
+  } = history;
+
+  useEffect(() => {
+    if (search && search.includes('data')) {
```
Comment (evgenijolshanskij) [inline modules/user/client-react/components/LoginView.web.jsx]: Not so reliable check.

File: modules/user/server-ts/social/googleExpo/resolvers.js (hunk 1 above)
Comment (BaloPhilip) [reply modules/user/server-ts/social/googleExpo/resolvers.js]: done

File: modules/authentication/client-react/helpers/index.js (hunk 2 above)
Comment (BaloPhilip) [reply modules/authentication/client-react/helpers/index.js]: done

File: modules/user/client-react/components/LoginView.web.jsx (hunk 4 above)
Comment (BaloPhilip) [reply modules/user/client-react/components/LoginView.web.jsx]: done

File: config/auth.js (hunk 3 above)
Comment (BaloPhilip) [reply config/auth.js]: done

File: modules/authentication/client-react/helpers/index.js [hunk 5]
```diff
@@ -1,12 +1,36 @@
...
+  const {
+    auth: {
+      social: {
+        [socialNetwork]: { enabled: isLogin },
+        [`${socialNetwork}Expo`]: { enabled: isExpoLogin }
+      }
+    }
+  } = settings;
+
+  if ((isLogin && isExpoLogin) || isLogin) {
```
Comment (evgenijolshanskij) [inline modules/authentication/client-react/helpers/index.js]: This condition doesn't make any sense for me.

File: modules/user/client-react/components/LoginForm.native.jsx [hunk 6]
```diff
@@ -15,29 +15,30 @@ const loginFormSchema = {
...
 };
-const { github, facebook, linkedin, google } = settings.auth.social;
+const { github, facebook, linkedin, google, googleExpo } = settings.auth.social;
 
 const renderSocialButtons = (buttonsLength, t) => {
   return buttonsLength > 2 ? (
     <View style={{ flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' }}>
       {facebook.enabled && <FacebookButton text={t('login.fbBtn')} type="icon" />}
-      {google.enabled && <GoogleButton text={t('login.googleBtn')} type="icon" />}
+      {(google.enabled || googleExpo.enabled) && <GoogleButton text={t('login.googleBtn')} type="icon" />}
```
Comment (evgenijolshanskij) [inline modules/user/client-react/components/LoginForm.native.jsx]: This solution is the cause of splitting `googleExpo` settings from the `google`.

File: modules/authentication/client-react/helpers/index.js (hunk 5 above)
Comment (evgenijolshanskij) [reply modules/authentication/client-react/helpers/index.js]: You just mixed up enabling `google` auth with the auth `type`.

Comment (BaloPhilip) [reply modules/authentication/client-react/helpers/index.js]: done

File: modules/user/client-react/components/LoginForm.native.jsx (hunk 6 above)
Comment (BaloPhilip) [reply modules/user/client-react/components/LoginForm.native.jsx]: gone

**Pull request closed without merging**